### PR TITLE
Embeds: Add Bluesky variation

### LINKED ISSUE #61349: Add Bluesky embed block variation
## What problem does this address?

Bluesky has just been added as a trusted oEmbed provider in core.

See https://core.trac.wordpress.org/changeset/58091 and https://core.trac.wordpress.org/ticket/61020

The Bluesky logo SVG can be found at https://github.com/bluesky-social/social-app/blob/main/bskyembed/assets/logo.svg / https://github.com/bluesky-social/social-app/blob/cdf7a1957ad6a5bd569f60fb7113c44973f4e2d7/bskyembed/assets/logo.svg

## What is your proposed solution?

Add a new embed block variation for Bluesky embeds for 6.6+

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?

Fixes #61349 

## Why?
- Bluesky has just been added as a trusted oEmbed provider in core so added it in the variation of embeds

## How?
- Adds a new variation for Bluesky embed

## Testing Instructions

1. Open a post or page.
2. Insert a Bluesky embed variation.

## Screenshots or screencast 
image

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

File: packages/block-library/src/embed/icons.js [hunk 1]
```diff
@@ -178,3 +178,12 @@ export const embedPocketCastsIcon = {
...
 	),
 };
+
+export const embedBlueskyIcon = (
+	<SVG xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 320 286">
+		<Path
+			fill="rgb(10,122,255)"
+			d="M69.364 19.146c36.687 27.806 76.147 84.186 90.636 114.439 14.489-30.253 53.948-86.633 90.636-114.439C277.107-.917 320-16.44 320 32.957c0 9.865-5.603 82.875-8.889 94.729-11.423 41.208-53.045 51.719-90.071 45.357 64.719 11.12 81.182 47.953 45.627 84.785-80 82.874-106.667-44.333-106.667-44.333s-26.667 127.207-106.667 44.333c-35.555-36.832-19.092-73.665 45.627-84.785-37.026 6.362-78.648-4.149-90.071-45.357C5.603 115.832 0 42.822 0 32.957 0-16.44 42.893-.917 69.364 19.147Z"
+		/>
+	</SVG>
```
Comment (t-hamano) [inline packages/block-library/src/embed/icons.js]:
```suggestion
	<SVG viewBox="0 0 24 24">
		<Path
			fill="#0a7aff"
			d="M6.3,4.2c2.3,1.7,4.8,5.3,5.7,7.2.9-1.9,3.4-5.4,5.7-7.2,1.7-1.3,4.3-2.2,4.3.9s-.4,5.2-.6,5.9c-.7,2.6-3.3,3.2-5.6,2.8,4,.7,5.1,3,2.9,5.3-5,5.2-6.7-2.8-6.7-2.8,0,0-1.7,8-6.7,2.8-2.2-2.3-1.2-4.6,2.9-5.3-2.3.4-4.9-.3-5.6-2.8-.2-.7-.6-5.3-.6-5.9,0-3.1,2.7-2.1,4.3-.9h0Z"
		/>
	</SVG>
```

The viewbox size of the other embedded icons is 24, so you might want to change this icon to 24 as well. Additionally, if the actual icon size is the same as the view box, it feels a bit cramped. I am changing the width to 20 like many other icons.

### Before
before

### After
after

File: packages/block-library/src/embed/variations.js [hunk 2]
```diff
@@ -360,6 +361,14 @@ const variations = [
 		patterns: [ /^https?:\/\/(www\.)?wolframcloud\.com\/obj\/.+/i ],
 		attributes: { providerNameSlug: 'wolfram-cloud', responsive: true },
 	},
+	{
+		name: 'bluesky',
+		title: 'Bluesky',
+		icon: embedBlueskyIcon,
+		description: __( 'Embed a Bluesky post.' ),
+		patterns: [ /^https?:\/\/(www\.)?bsky\.app\/obj\/.+/i ],
+		attributes: { providerNameSlug: 'bluesky', responsive: true },
```
Comment (t-hamano) [inline packages/block-library/src/embed/variations.js]:
```suggestion
		attributes: { providerNameSlug: 'bluesky' },
```

When the `responsive` property is `true`, a control appears in the block sidebar that determines whether to keep the aspect ratio when the browser is resized.

responsive-setting

Bluesky changes the aspect ratio depending on the content, so I think this setting is probably unnecessary.

**Review by t-hamano — CHANGES_REQUESTED** [review]
Thanks for the PR!

I left two suggestions for this PR, so I would appreciate it if you could check them out.

**Comment by amitraj2203** [comment]
Hi @t-hamano, I agree with your suggestions and have reworked on the PR.

Thank You.

**Comment by t-hamano** [comment]
LGTM 🚀

image

**Review by t-hamano — APPROVED** [review]

**Review by t-hamano — CHANGES_REQUESTED** [review]
Oh wait, it looks like the pattern regular expression is incorrect.

File: packages/block-library/src/embed/variations.js [hunk 3]
```diff
@@ -360,6 +361,14 @@ const variations = [
 		patterns: [ /^https?:\/\/(www\.)?wolframcloud\.com\/obj\/.+/i ],
 		attributes: { providerNameSlug: 'wolfram-cloud', responsive: true },
 	},
+	{
+		name: 'bluesky',
+		title: 'Bluesky',
+		icon: embedBlueskyIcon,
+		description: __( 'Embed a Bluesky post.' ),
+		patterns: [ /^https?:\/\/(www\.)?bsky\.app\/obj\/.+/i ],
```
Comment (t-hamano) [inline packages/block-library/src/embed/variations.js]:
```suggestion
		patterns: [ /^https?:\/\/bsky\.app\/profile\/.+\/post\/.+/i ],
```

Looking at [the core changeset](https://core.trac.wordpress.org/changeset/58091), isn't this the pattern?

cc @swissspidy

Comment (amitraj2203) [reply packages/block-library/src/embed/variations.js]: @t-hamano Thank you for catching that! I've updated it to match the correct pattern now. Appreciate your attention to detail!

**Comment by t-hamano** [comment]
Thanks for the update!

Could you rebase your branch with the latest trunk again? This should fix the failing CI check.

**Comment by amitraj2203** [comment]
> Thanks for the update!
> 
> Could you rebase your branch with the latest trunk again? This should fix the failing CI check.

Hi @t-hamano, I've rebased my branch with the latest trunk. Everything looks good except for one failing CI check.

**Comment by t-hamano** [comment]
Thanks for the Update!

Since Windows job failure is a known issue, I would like to merge this PR ([Slack link](https://wordpress.slack.com/archives/C02QB2JS7/p1714590550533979)).

**Review by t-hamano — APPROVED** [review]

**Pull request merged**